[MIG] base_fontawesome: Migration to 11.0

etobella opened the pull request:
This module upgrades the fontawesome font to 5.3.1.
It allows to use new icons.
Following the instructions of OCA/web#1042

simahawk committed:
[add] base_fontawesome (#866)

simahawk committed:
[MIG][10.0] base_fontawesome

oca-travis committed:
[UPD] Update base_fontawesome.pot

etobella committed:
[MIG] base_fontawesome: Migration to 11.0

AaronHForgeFlow commented:
Got this on runbot when logging in, no idea if it is due this module:
![image](https://user-images.githubusercontent.com/19620251/45758166-44ddd280-bc25-11e8-8a5c-df32c27a545f.png)

etobella commented:
The gitignore was removing the lib folder. I removed it, following the configuration of OCA/web

JordiBForgeFlow reviewed (approved):
:+1: Functional review. Works great!

JordiBForgeFlow commented on base_fontawesome/README.rst:
```diff
@@ -0,0 +1,59 @@
+* brand_icon
+The usage depends on the configuration of the icon.
+
+For example, brand_icon is necessary if we are using an icon of a branch::
```
brand, not branch

LoisRForgeFlow reviewed (approved):
Tested on runbot, looks great!

AaronHForgeFlow reviewed (approved):
:+1: Functional test

JordiBForgeFlow commented:
@simahawk we can wait for your final review, as you authored the module in v10, and then this one can be merged.

simahawk commented on base_fontawesome/templates/assets.xml:
```diff
@@ -0,0 +1,24 @@
+
+	<template id="assets_backend" inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+			<link rel="stylesheet" type="text/css" href="/base_fontawesome/static/lib/fontawesome-5.3.1/css/all.css" />
```
can we get rid of unused resources? :)

etobella replied:
If I replace the elements it is giving problems, related to the usage of FontAwesome font directly on some code by odoo. I will try to review if it is possible to fix the problem.

simahawk commented on base_fontawesome/templates/assets.xml:
```diff
@@ -0,0 +1,24 @@
+-->
+<odoo>
+
+	<template id="assets_backend" inherit_id="web.assets_backend">
```
AFAIK you should use assets_common to match both back and frontend

simahawk commented on base_fontawesome/__manifest__.py:
```diff
@@ -0,0 +1,17 @@
+{
+    'name': 'Base Fontawesome',
+    'summary': """Up to date Fontawesome resources.""",
+    'version': '11.0.1.0.0',
```
module version, as per README, should match the version of FA ;)

etobella commented:
@simahawk I finally applied all the changes that you suggested. There was an issue with the configuration of some buttons, but it seems right now. I also changed the version.

Just in order to leave it prepared for merge, I rebased and squashed all the fixup comits.

simahawk commented on base_fontawesome/static/lib/fontawesome-5.3.1/css/brands.css:
```diff
@@ -0,0 +1,13 @@
+/*!
```
already mentioned in the old PR: do we really have to keep both compressed and uncompressed?
Our repos are already heavy... I think it's better to drop useless files especially if big :)

etobella replied:
I will remove all the unused files.

simahawk commented on base_fontawesome/templates/assets.xml:
```diff
@@ -0,0 +1,27 @@
+<odoo>
+
+	<template id="assets_common" inherit_id="web.assets_common">
+		<xpath expr="link[@href='/web/static/lib/fontawesome/css/font-awesome.css']" position="replace">
```
can you check indentation? seems messed up :/

simahawk reviewed (approved):
LG thanks!

simahawk commented:
you can squash & merge sir :)

etobella commented:
Well, I found an small issue on sortable columns. I will fix it first :wink:

etobella commented:
![imagen](https://user-images.githubusercontent.com/28590170/45879755-f3a91c80-bda5-11e8-9dca-e7c1c1d68f5f.png)
Sorting icons are not shown correctly, because It tries to find FontAwesome font but it no longer exists. 
We cannot replace fontawesome original css...

simahawk commented:
loading them all does not sound like a good idea, sorry :)
Who's causing the issue? Who tries to find what?
Where is the line adding that action? Can't we make it use new FA?

etobella commented:
The problem is that the old font is referenced on some lessc files:
https://github.com/odoo/odoo/search?l=Less&q=FontAwesome
As they are using it directly it gives problems.

simahawk commented:
ahh they've changed the name....... `font-family: "Font Awesome 5 Free";  /* updated font-family */`

simahawk commented:
@etobella do you think we can simply re-define the font-face w/ the old name?
Like here https://github.com/OCA/server-tools/pull/1361/files#diff-5163cade13025e35cf7b4df7687f8697R3787
We could split it to a different compat-only file

etobella commented:
I finally made it!!
![imagen](https://user-images.githubusercontent.com/28590170/45883045-3cfe6980-bdb0-11e8-963d-9687e2c3c3c7.png)
The problem was related to the link of the font :)
